src/geomath.rs

```diff
@@ -291,66 +287,63 @@ pub fn astroid(x: f64, y: f64) -> f64 {
     }
 }
 
-pub fn _A1m1f(eps: f64, geodesic_order: i64) -> f64 {
+pub fn _A1m1f(eps: f64, geodesic_order: usize) -> f64 {
     const COEFF: [f64; 5] = [1.0, 4.0, 64.0, 0.0, 256.0];
-    let m: i64 = geodesic_order / 2;
-    let t = polyval(m as isize, &COEFF, sq(eps)) / COEFF[(m + 1) as usize];
+    let m = geodesic_order / 2;
+    let t = polyval(m, &COEFF, sq(eps)) / COEFF[m + 1];
     (t + eps) / (1.0 - eps)
 }
 
-pub fn _C1f(eps: f64, c: &mut [f64], geodesic_order: i64) {
+pub fn _C1f(eps: f64, c: &mut [f64], geodesic_order: usize) {
     const COEFF: [f64; 18] = [
         -1.0, 6.0, -16.0, 32.0, -9.0, 64.0, -128.0, 2048.0, 9.0, -16.0, 768.0, 3.0, -5.0, 512.0,
         -7.0, 1280.0, -7.0, 2048.0,
     ];
     let eps2 = sq(eps);
     let mut d = eps;
     let mut o = 0;
-    for l in 1..=geodesic_order {
+    for (l, v) in c.iter_mut().enumerate().take(geodesic_order + 1).skip(1) {
```

Comment:
I'm not sure we can really consider this line an improvement in aggregate, but who am I to argue with a paperclip? 😆